lib/config/fileconf.go

```diff
@@ -420,9 +414,6 @@ func (conf *FileConfig) CheckAndSetDefaults() error {
 			return trace.BadParameter("MAC algorithm %q is not supported; supported algorithms: %q", m, sc.MACs)
 		}
 	}
-	if conf.CASignatureAlgorithm != nil && !apiutils.SliceContainsStr(validCASigAlgos, *conf.CASignatureAlgorithm) {
```

Comment:
Let's add a deprecation notice at fileconf.go / Global.CASignatureAlgorithm too, so it's clear the field is unused and kept only for config backwards compatibility.

Is there a "Cloud"/protobuf counterpart for this setting, or just the fileconf / teleport.yaml one?

Reply:
Deprecation notice added :pray: 

> Is there a "Cloud"/protobuf counterpart for this setting, or just the fileconf / teleport.yaml one?

CertAuthorityV2 has two methods that I'm pretty sure could be removed: `Get/SetSigningAlg` 

My concern is having a version without those methods interacting with a version that has them
I don't know how that works out (backwards/forward compat)
I'll do some tests :hammer_and_wrench:

Reply:
If you drop the methods and an older gRPC client tries to call them, it'll get a framework-level error saying the method doesn't exist (I don't remember the exact error, but it's something like that).

Whether it's safe to drop depends on how it's used. If you can ensure all calls are removed by version N, then it's safe to drop the method in N+1.